baserun/annotation.py

```diff
@@ -124,10 +124,10 @@ def check(
         self,
         name: str,
         methodology: str,
-        expected: dict[str, Any],
-        actual: dict[str, Any],
+        expected: Dict[str, Any],
```

Comment:
It's my understanding that using the built-in types versus the generics in `typing` is preferred. 

https://peps.python.org/pep-0585/
https://discuss.python.org/t/concern-about-pep-585-removals/15901

Reply:
sure is preferred. however, you can't say that baserun supports python 3.8 while it's actually not. you need to keep consistent about it and either not use python 3.9+ style for such type hints or mark a library as 3.9+ compatible

Reply:
Ah yeah good point, I forgot that this was a new thing. I'll change it globally in a subsequent PR